core/src/main/java/org/elasticsearch/index/Index.java

```diff
@@ -31,21 +32,23 @@
 public class Index implements Streamable {
 
     private String name;
```

Comment:
also make sure these are both final and impl Writeable as we are at it? I did that before it's not too much of an overhead